sklearn/compose/tests/test_column_transformer.py

```diff
@@ -94,41 +94,61 @@ def test_column_transformer():
     ]
 
     for selection, res in cases:
-        ct = ColumnTransformer([('trans', Trans(), selection)],
+        ct = ColumnTransformer([('trans', selection, Trans())],
                                remainder='drop')
         assert_array_equal(ct.fit_transform(X_array), res)
         assert_array_equal(ct.fit(X_array).transform(X_array), res)
 
         # callable that returns any of the allowed specifiers
-        ct = ColumnTransformer([('trans', Trans(), lambda x: selection)],
+        ct = ColumnTransformer([('trans', lambda x: selection, Trans())],
                                remainder='drop')
         assert_array_equal(ct.fit_transform(X_array), res)
         assert_array_equal(ct.fit(X_array).transform(X_array), res)
 
-    ct = ColumnTransformer([('trans1', Trans(), [0]),
-                            ('trans2', Trans(), [1])])
+    ct = ColumnTransformer([('trans1', [0], Trans()),
+                            ('trans2', [1], Trans())])
     assert_array_equal(ct.fit_transform(X_array), X_res_both)
     assert_array_equal(ct.fit(X_array).transform(X_array), X_res_both)
     assert len(ct.transformers_) == 2
 
     # test with transformer_weights
     transformer_weights = {'trans1': .1, 'trans2': 10}
-    both = ColumnTransformer([('trans1', Trans(), [0]),
-                              ('trans2', Trans(), [1])],
+    both = ColumnTransformer([('trans1', [0], Trans()),
+                              ('trans2', [1], Trans())],
                              transformer_weights=transformer_weights)
     res = np.vstack([transformer_weights['trans1'] * X_res_first1D,
                      transformer_weights['trans2'] * X_res_second1D]).T
     assert_array_equal(both.fit_transform(X_array), res)
     assert_array_equal(both.fit(X_array).transform(X_array), res)
     assert len(both.transformers_) == 2
 
-    both = ColumnTransformer([('trans', Trans(), [0, 1])],
+    both = ColumnTransformer([('trans', [0, 1], Trans())],
                              transformer_weights={'trans': .1})
     assert_array_equal(both.fit_transform(X_array), 0.1 * X_res_both)
     assert_array_equal(both.fit(X_array).transform(X_array), 0.1 * X_res_both)
     assert len(both.transformers_) == 1
 
 
+@pytest.mark.parametrize("t", [('drop', [0]),
```

Comment:
maybe I'm misreading but why is ``[0]`` a parameter?
Also can you add a test where the first transformer is ``('somename', 'passthrough', 'passthrough')``?

Reply:
the `[0]` was there cause initially I wanted to try the case you just mentioned. But the `ColumnTransformer` doesn't accept an integer or a string literal, it has to be a list of columns, so it would have to be `('somename', ['passthrough'], 'passthrough')` which makes is less exciting and more explicit about which one's the transformer.